# Malaria Project

This project investigates evolutionary and structural features of *Plasmodium falciparum* proteins using sequence alignment, conservation analysis, and phylogenetic trees. It was developed as part of the **Proteomes, Interactomes, and Biological Networks** course (AA 2024â€“2025) at the University of Bologna.

## Repository Contents

### Notebooks

| Notebook | Description |
|----------|-------------|
| `conserved_regions.ipynb` | Builds a phylogenetic tree using **IQ-TREE** and analyzes conserved regions. Includes plots of the results. |
| `matrices.ipynb` | Creates distance matrices and analyses cluster |
| `tree_v2.ipynb` | An alternative version of the phylogenetic tree visualization using **IQ-TREE**; optimized for 4-strain plots. |

### Tools and Methods
- **IQ-TREE**: For constructing phylogenetic trees from aligned sequences.
- **Python**: Data manipulation, visualization, and matrix generation.

### How to Run

You can open any notebook directly in Google Colab:

- [Conserved Regions](https://colab.research.google.com/github/Sivyan/Malaria-project/blob/main/conserved_regions.ipynb)
- [Tree Visualization (v2)](https://colab.research.google.com/github/Sivyan/Malaria-project/blob/main/tree_v2.ipynb)
- [Matrices Notebook](https://colab.research.google.com/github/Sivyan/Malaria-project/blob/main/matrices.ipynb)

## Course Context

This work is part of a final team project requiring use of at least 4 resources, formats, or tools such as:
- PDB, UniProt, Ensembl
- FASTA, PDB, DSSP formats
- ChimeraX, PyMOL, Cytoscape, NetworkX

